keras_nlp/utils/text_generation.py

```diff
@@ -109,16 +152,111 @@ def token_probability_fn(inputs):
         i += 1
 
     if end_token_id is not None:
-        # Mask out tokens after `end_token_id` is encountered.
-        # Find index of first end_token_id.
-        end_indices = tf.math.argmax(prompt == end_token_id, -1)
-        # Use max_length if no `end_token_id` is found.
-        end_indices = tf.where(end_indices == 0, max_length, end_indices)
-        # Build a mask including end_token and replace tokens after end_token
-        # with `pad_token_id`.
-        valid_indices = tf.sequence_mask(end_indices + 1, maxlen=max_length)
-        prompt = tf.where(valid_indices, prompt, pad_token_id)
+        prompt = mask_tokens_after_end_token(
+            prompt, max_length, end_token_id, pad_token_id
+        )
+
+    if input_is_1d:
+        return tf.squeeze(prompt)
+    return prompt
+
+
+def random_search(
```

Comment:
We need to add this to the `__init__.py` for the utils dir, so this gets exported.